tools/auth/twitter.go

```diff
@@ -0,0 +1,49 @@
+package auth
+
+import (
+	"golang.org/x/oauth2"
+)
+
+var _ Provider = (*Twitter)(nil)
+
+// NameTwitter is the unique name of the Twitter provider.
+const NameTwitter string = "twitter"
+
+// Twitter allows authentication via Twitter OAuth2.
+type Twitter struct {
+	*baseProvider
+}
+
+// NewTwitterProvider creates new Twitter provider instance with some defaults.
+func NewTwitterProvider() *Twitter {
+	return &Twitter{&baseProvider{
+		scopes:     []string{"users.read"},
+		authUrl:    "https://twitter.com/i/oauth2/authorize",
+		tokenUrl:   "https://api.twitter.com/2/oauth2/token",
+		userApiUrl: "https://api.twitter.com/1.1/account/verify_credentials.json?include_email=true",
+	}}
+}
+
+// FetchAuthUser returns an AuthUser instance based on the Twitter's user api.
+func (p *Twitter) FetchAuthUser(token *oauth2.Token) (*AuthUser, error) {
+	// https://developer.twitter.com/en/docs/twitter-api/users/lookup/api-reference/get-users-me
+	rawData := struct {
+		Id                   string
+		Name                 string
+		Email                string
+		ProfileImageUrlHttps string
```

Comment:
`ProfileImageUrlHttps` will not be unmarshalized because in the [twitter response](https://developer.twitter.com/en/docs/twitter-api/v1/accounts-and-users/manage-account-settings/api-reference/get-account-verify_credentials#example-response) it is returned as `profile_image_url_https`.

Please add json tags to the struct members mapped to the response fields (id, name and email including for consistency).